### PR TITLE
quality of life improvements

### PR DESCRIPTION
**Pull request by infinitewarp**
This PR does a few things:

* Clean up the bash scripts and fix lots of potential problems (like unquoted variables) spotted by `shellcheck` and `shfmt`.
* Add a GitHub workflow job to run `shellcheck` and `shfmt`. I intend to set this as a required job after this PR merges. I forced both to fail by pushing known-bad commits and watching the jobs fail.
* Move the `*.env` files from `./config/` to `./env/` so someone like me can run the installer directly from this checked-out repo, not just the RPMs, by setting the `DISCOVERY_PKG_ROOT` environment variable.
* Do not require the user to enter a value for the Django secret. Add option to generate one automatically.
* Do not require the user to run separate `create-server-password` and `create-app-secret` commands before `install`. Invoke them directly inline only if needed.

I confirmed that everything still seems to work after these changes. On a Fedora 40 system, I repeatedly installed and uninstalled the files and stopped and started the systemd units. I also placed a custom `discovery-db.container` and `env-db.env` in an "overrides" directory, invoked the installer with `--override-conf-dir`, and confirmed that the changes were made and the containers came up as expected. (Specifically, I changed the postgresql image to 16 (default is 15) and changed the psql username and database name.)

If we disagree with some of these changes, I can drop and cherry-pick them into separate branches for further review and development.

### CODE REVIEW

File: bin/create-app-secret [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+      prompt_key
+      exit
+      ;;
+    esac
+  done
+}
+
+while getopts "Yy" opt; do
+  case $opt in
+  [Yy])
```
Comment (abellotti) [inline bin/create-app-secret:50]: usage below around line 254 should be updated to mention this.

echo "  create-app-secret  [-y]      - Create a Discovery application secret"

File: bin/create-app-secret [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+  if [ ! "${discovery_django_secret_key}" = "${discovery_django_secret_key2}" ]; then
+    echo "Discovery Django secret keys do not match."
+    exit 1
+  fi
+  set_django_secret_key "${discovery_django_secret_key}"
+}
+
+function prompt_for_generation {
+  while true; do
+    read -rp "Automatically generate new Django secret key? [y/n]: " yn
```
Comment (abellotti) [inline bin/create-app-secret:34]: will need minor update to README.md and https://gitlab.cee.redhat.com/discovery/unofficial-internal-documentation/-/blob/main/how-to-guides/Deployments/Quadlet.md

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+      - run: go install mvdan.cc/sh/v3/cmd/shfmt@latest
+      - run: sudo apt-get install -y shellcheck
+      - name: run shellcheck
+        run: shellcheck ./bin/*
+      - name: run shfmt
+        run: shfmt -i 2 --diff ./bin/*
```
Comment (abellotti) [inline .github/workflows/lint.yml:20]: 👍  very nice to have.   Maybe not in this PR but would be nice to have a Makefile so we can do `make lint` before we push stuff.

File: bin/create-app-secret (hunk 1 above)
Comment (infinitewarp) [reply bin/create-app-secret:50]: Adding `-y` to run `discovery-installer create-app-secret -y` doesn't actually pass that extra argument along here in `discovery-installer`. It's only accepted if you invoke the `create-app-secret` script _directly_. So, for now, that docstring is still _technically_ correct.

We could expose it here pretty easily, but the existing argument parsing is already so spread out (hundreds of lines between `shift`s 😱), I'd prefer to rewrite and centralize that logic for readability and flexibility. However, I'd also rather punt any such rewrite to `[unspecified future date]` when we evolve this code into something of a `discoveryctl` which will hopefully have a _much more robust_ arg parser, whether that's the `getopt` program or Python's `argparse` module. 😉

File: bin/create-app-secret (hunk 2 above)
Comment (infinitewarp) [reply bin/create-app-secret:34]: https://gitlab.cee.redhat.com/discovery/unofficial-internal-documentation/-/merge_requests/52

File: .github/workflows/lint.yml (hunk 3 above)
Comment (infinitewarp) [reply .github/workflows/lint.yml:20]: Good idea! Added in the latest commit.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.PHONY: help all lint
+.DEFAULT_GOAL := all
```
Comment (abellotti) [inline Makefile:2]: Minor, but I'd change DEFAULT_GOAL to help.

**Review by abellotti — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (infinitewarp) [reply Makefile]: Done!

**Pull request merged**